File: Hashmap/Ransom Note/RansomNote.cs
public class Solution {
    public bool CanConstruct(string ransomNote, string magazine) {
        Dictionary<char, int> charCount = new Dictionary<char, int>();
        foreach (char word in magazine) {
            if (charCount.ContainsKey(word)) {
                charCount[word]++;
            } else {
                charCount[word] = 1;
            }
        }

        foreach (char word in ransomNote) {
            if (charCount.ContainsKey(word)) {
                charCount[word]--;
            } else {
                charCount[word] = -1;
            }
        }

        foreach (var entry in charCount) {
            if (entry.Value < 0) {
                return false;
            }
        }

        return true;
    }
}

File: Hashmap/Ransom Note/RansomNote.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        unordered_map<char, int> charCount;
        for (char word : magazine)
            charCount[word]++;

        for (char word : ransomNote)
            charCount[word]--;

        for (const auto& entry : charCount) {
            if (entry.second < 0)                
                return false;
        }
        return true;
    }
};